### PR TITLE
feat: Unify all light bridge services into one 

### PR DESCRIPTION
**Pull request by wsdt**
# Changes
- Unifying LightBridge service
- Airdrop config moved to BobaChain map with optional configuration via Lightbridge options

## Tests
- Existing tests adapted
- Parallization tests added
- SourceChainId check to make arbitrage attacks on the airdrop unattractive.

Please carefully review. 

## Outstanding to cover in separate issue
[ ] UX Adaption on Gateway to only show routes that are supported: https://github.com/bobanetwork/gateway/pull/380

### CODE REVIEW

File: src/service.ts [hunk 1]
```diff
@@ -113,23 +109,26 @@ export class LightBridgeService extends BaseService<TeleportationOptions> {
...
-      this.options.ownSupportedAssets
-    ).find(
-      (k) => this.options.ownSupportedAssets[k?.toLowerCase()] === Asset.BOBA
-    )
-    if (!bobaTokenContractAddr) {
+    const defaultAssetAddr =
+      Object.keys(this.options.ownSupportedAssets).find(
+        (k) => this.options.ownSupportedAssets[k?.toLowerCase()] === Asset.BOBA
+      ) ?? '0x0000000000000000000000000000000000000000'
+    if (!defaultAssetAddr) {
```
Comment (boyuan-chen) [inline src/service.ts:126]: The default value is `0x0000000000000000000000000000000000000000`, so the `if` is always false. Is it right?

**Review by boyuan-chen — APPROVED** [review]

File: src/service.ts [hunk 2]
```diff
@@ -400,42 +403,73 @@ export class LightBridgeService extends BaseService<TeleportationOptions> {
...
   async _fulfillsAirdropConditions(disbursement: Disbursement) {
     const nativeBalance = await this.state.Teleportation.provider.getBalance(
       disbursement.addr
     )
-    if (nativeBalance.gt(this.options.airdropConfig.airdropAmountWei)) {
+    const sourceLayer: ELayer =
+      this.state.supportedChains.find(
+        (c) => c.chainId.toString() === disbursement.sourceChainId.toString()
+      )?.layer ?? BobaChains[disbursement.sourceChainId]?.layer
+    if (sourceLayer === ELayer.Layer2) {
```
Comment (souradeep-das) [inline src/service.ts:448]: but this does not prevent the case where bnb is the source?

Comment (wsdt) [reply src/service.ts]: exactly, changed now to only ETH!

File: src/exec/run.ts [hunk 3]
```diff
@@ -45,123 +42,116 @@ const main = async () => {
...
+      awsKmsAccessKey,
+      awsKmsEndpoint,
+      awsKmsKeyId,
+      awsKmsRegion,
+      awsKmsSecretKey,
     },
-    []
+  }
+
+  const isTestnetMode = ENetworkMode.TESTNETS === networkMode
```
[the remark this answers is not in the document]
Comment (wsdt) [reply src/exec/run.ts:121]: done!

File: src/service.ts [hunk 4]
```diff
@@ -400,42 +403,73 @@ export class LightBridgeService extends BaseService<TeleportationOptions> {
...
     const nativeBalance = await this.state.Teleportation.provider.getBalance(
       disbursement.addr
     )
-    if (nativeBalance.gt(this.options.airdropConfig.airdropAmountWei)) {
+    const sourceLayer: EAirdropSource =
+      this.state.supportedChains.find(
+        (c) => c.chainId.toString() === disbursement.sourceChainId.toString()
+      )?.layer ?? BobaChains[disbursement.sourceChainId]?.layer
+    if (sourceLayer === EAirdropSource.PROHIBIT) {
+      this.logger.info(`Not airdropping as sourceNetwork is a L2.`, {
```
Comment (souradeep-das) [inline src/service.ts:449]: minor
```suggestion
      this.logger.info(`Not airdropping as sourceNetwork is prohibited.`, {
```

**Review by souradeep-das — APPROVED** [review]

**Pull request merged**